internal/config/config.go

```diff
@@ -1,8 +1,108 @@
 package config
 
-import "github.com/spf13/cobra"
+import (
+	"os"
 
-type Config interface {
-	Init(cmd *cobra.Command) error
-	Set()
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+type Root struct {
+	Debug   bool
+	CfgFile string
+}
+
+func (Root) Init(cmd *cobra.Command) error {
+	cmd.PersistentFlags().BoolP("debug", "d", false, "enable debug mode")
+	if err := viper.BindPFlag("debug", cmd.PersistentFlags().Lookup("debug")); err != nil {
+		return err
+	}
+
+	cmd.PersistentFlags().String("config", "", "configuration file path")
+	if err := viper.BindPFlag("config", cmd.PersistentFlags().Lookup("config")); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *Root) Set() {
+	s.Debug = viper.GetBool("debug")
+	s.CfgFile = viper.GetString("config")
+}
+
+type Server struct {
+	Cert   string
+	Key    string
+	Bind   string
+	Static string
+	Proxy  bool
+
+	BaseDir  string            `yaml:"basedir,omitempty"`
+	Streams  map[string]string `yaml:"streams"`
+	Profiles string            `yaml:"profiles,omitempty"`
+}
```

Comment:
In the future, I would like to split HLS, HTTP (live) and on-demand streaming to different modules. And every module should have separate config. And they would be independent. That's why i think, for future, it would be better to have split config.

Reply:
moved #7